apps/discord-bot-internal/src/services/spy.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  Colors,
+  EmbedBuilder,
+  Events,
+  GuildMember,
+  GuildTextBasedChannel,
+  Message,
+  PartialMessage,
+  ReadonlyCollection
+} from 'discord.js';
+import { Service } from '../types/service';
+import { config } from '../config';
+import { timeSpanToPrettyPrint } from '../utils';
+import { MomentumColor } from '../momentum-color';
+
+export class SpyService extends Service {
+  private messageLogChannel?: GuildTextBasedChannel;
+  private joinLogChannel?: GuildTextBasedChannel;
+
+  async init() {
+    this.messageLogChannel = await this.client.channels
+      .fetch(config.message_history_channel)
+      .then((ch) =>
+        ch?.isTextBased() ? (ch as GuildTextBasedChannel) : undefined
+      );
+
+    this.joinLogChannel = await this.client.channels
+      .fetch(config.join_log_channel)
+      .then((ch) =>
+        ch?.isTextBased() ? (ch as GuildTextBasedChannel) : undefined
+      );
+
+    if (!this.messageLogChannel || !this.joinLogChannel) {
+      throw new Error(
+        `Failed to initialize spy module: text channel ${
+          !this.messageLogChannel
+            ? config.message_history_channel
+            : config.join_log_channel
+        } is not found`
+      );
+    }
+
+    this.client.on(Events.GuildMemberAdd, this.guildMemeberAdd.bind(this));
+    this.client.on(Events.MessageDelete, this.messageDelete.bind(this));
+    this.client.on(Events.MessageUpdate, this.messageUpdate.bind(this));
+    this.client.on(Events.MessageBulkDelete, this.messageBulkDelete.bind(this));
+  }
+
+  async guildMemeberAdd(member: GuildMember) {
```

Comment:
`Memeber` (sounds like a name Elon Musk would suggest naming a month to when off his face on ketamine)